src/libraries/System.Private.CoreLib/src/System/Globalization/Calendar.cs

```diff
@@ -140,7 +127,7 @@ internal static void CheckAddResult(long ticks, DateTime minValue, DateTime maxV
             }
         }
 
-        internal DateTime Add(DateTime time, double value, int scale)
```

Comment:
This is the one non-trivial change. Because we are now passing `long` constants for the scale, this could be a tiny performance regression in the `value * scale` calculation on [line 152](https://github.com/dotnet/runtime/pull/103498/files#diff-8c1794ccb5af91ff931a812d0a8c888bc85f12b07535ffc0119fec742c29a8c9L152) where previously it was a `double * int` it's now a `double * long`.

Reply:
@EgorBo what do you think about this change? 

If we need to avoid that we can cast down the input scale to `int` and assert for the input value is inside the `int` range.

Reply:
To clarify, we call the `internal DateTime Add(` in five places in \dotnet\runtime\src\libraries\System.Private.CoreLib\src\System\Globalization\Calendar.cs	
					
Line 160:	return Add(time, milliseconds, 1);
Line 172:	return Add(time, days, TimeSpan.MillisecondsPerDay);
Line 184:	return Add(time, hours, TimeSpan.MillisecondsPerHour);
Line 196: 	return Add(time, minutes, TimeSpan.MillisecondsPerMinute);
Line 228:	return Add(time, seconds, TimeSpan.MillisecondsPerSecond);

All those constants are small enough to fit an `int` (and could be, but see original Issue, but for safety sake, we left them as `long`s). In EACH of _these_ uses, the `long` constant _will_ fit in an `int`, so I can put this line back to accepting an `int` parameter (and cast to int at the five call sites) and it will compile fine.

Any preference of style?

Reply:
Either casting the values in the calling sites or in the method itself is fine with me.

Reply:
I'm leaning toward casting at call sites. Pushed that change